WeatherMonitoringAndReportingService/Logic/Readers/JsonFormatReader.cs

```
@@ -0,0 +1,20 @@
+﻿using System.Text.Json;
+using WeatherMonitoringAndReportingService.Logic.Models;
+
+namespace WeatherMonitoringAndReportingService.Logic.Readers
+{
+    public class JsonFormatReader : IFormatReader
+    {
+        public WeatherData? ParseWeatherData(string userInput)
+        {
+            try
+            {
+                return JsonSerializer.Deserialize<WeatherData>(userInput);
+            }
+            catch (Exception ex)
```

we need to specify the exception that related to invalid format, and then we shouldn't just swallow the exception and return null, we should throw an appropriate exception indicating invalid exception